src/js/node/http2.ts

```diff
@@ -3354,10 +3354,32 @@ class ClientHttp2Session extends Http2Session {
         }
         headers[":scheme"] = scheme;
       }
+
       if (headers[":path"] == undefined) {
         headers[":path"] = "/";
       }
 
+      if (options !== undefined) {
+        if (!$isObject(options) || $isArray(options)) {
+          throw $ERR_INVALID_ARG_TYPE("options", "object", options);
+        }
+        if (options.endStream !== undefined && typeof options.endStream !== "boolean") {
+          throw $ERR_INVALID_ARG_TYPE("options.endStream", "boolean", options.endStream);
+        }
+        if (options.weight !== undefined && typeof options.weight !== "number") {
+          throw $ERR_INVALID_ARG_TYPE("options.weight", "number", options.weight);
+        }
+        if (options.parent !== undefined && typeof options.parent !== "number") {
+          throw $ERR_INVALID_ARG_TYPE("options.parent", "number", options.parent);
+        }
+        if (options.exclusive !== undefined && typeof options.exclusive !== "boolean") {
+          throw $ERR_INVALID_ARG_TYPE("options.exclusive", "boolean", options.exclusive);
+        }
+        if (options.silent !== undefined && typeof options.silent !== "boolean") {
+          throw $ERR_INVALID_ARG_TYPE("options.silent", "boolean", options.silent);
```

Comment:
should these also set the default values from `setAndValidatePriorityOptions`? https://github.com/nodejs/node/blob/1b5b12c3e6212d79884ae328c5710d4c9d2c2d8c/lib/internal/http2/core.js#L786

Reply:
Good catch thanks, will investigate

Reply:
@cirospaciari might know. it's possible we set them later